--- spcaview-20061208.plain/Makefile	2006-12-19 19:02:58.000000000 +0000
+++ spcaview-20061208/Makefile	2006-12-19 19:03:24.000000000 +0000
@@ -8,8 +8,9 @@
 APP_BINARY=spcaview
 BIN=/usr/local/bin
 #SDLLIBS= -lSDL -lpthread -lSDL_image
-SDLLIBS = $(shell sdl-config --libs) 
-SDLFLAGS = $(shell sdl-config --cflags)
+SDL_CONFIG = sdl-config
+SDLLIBS = $(shell $(SDL_CONFIG) --libs) 
+SDLFLAGS = $(shell $(SDL_CONFIG) --cflags)
 
 SERVFLAGS= -O2 -DLINUX $(WARNINGS)
 MATH_LIB=-lm 
