--- spcaview-20061208.plain/Makefile	2006-12-19 18:31:35.000000000 +0000
+++ spcaview-20061208/Makefile	2006-12-19 18:31:53.000000000 +0000
@@ -57,10 +57,10 @@
 	chmod 755 $(APP_BINARY)
 
 spcaserv: $(OBJSERVER)
-	gcc $(SERVFLAGS) -o spcaserv $(OBJSERVER) $(SERVLIBS)
+	$(CC) $(SERVFLAGS) -o spcaserv $(OBJSERVER) $(SERVLIBS)
 	
 spcacat: $(OBJCAT)
-	gcc $(SERVFLAGS) -o spcacat $(OBJCAT) $(SERVLIBS)
+	$(CC) $(SERVFLAGS) -o spcacat $(OBJCAT) $(SERVLIBS)
 	
 spcaview.o: spcaview.c  jconfig.h dpsh.h utils.h SDL_audioin.h
 
