--- uClibc-0.9.28/libc/sysdeps/linux/arm/ioperm.c.orig	2006-12-14 23:06:37.000000000 +0000
+++ uClibc-0.9.28/libc/sysdeps/linux/arm/ioperm.c	2006-12-14 23:08:27.000000000 +0000
@@ -105,13 +105,8 @@
     char systype[256];
     int i, n;
     
-#if LINUX_VERSION_CODE < 132119
-    static int iobase_name[] = { CTL_BUS, BUS_ISA, BUS_ISA_PORT_BASE };
-    static int ioshift_name[] = { CTL_BUS, BUS_ISA, BUS_ISA_PORT_SHIFT };
-#else
     static int iobase_name[] = { CTL_BUS, CTL_BUS_ISA, BUS_ISA_PORT_BASE };
     static int ioshift_name[] = { CTL_BUS, CTL_BUS_ISA, BUS_ISA_PORT_SHIFT };
-#endif
 
     size_t len = sizeof(io.base);
 
