--- ixp425-eth/Makefile	2006-12-23 01:43:52.000000000 +0000
+++ ixp425-eth.patched/Makefile	2006-12-23 01:42:00.000000000 +0000
@@ -16,6 +16,7 @@
           -I$(OSAL_DIR)/os/linux/include/core/  \
           -I$(OSAL_DIR)/os/linux/include/platforms/ \
           -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ixp425 \
           -I$(OSAL_DIR)/os/linux/include/core/ \
           -I$(OSAL_DIR)/include/ \
           -I$(OSAL_DIR)/include/modules/ \
