--- linux-2.6.15/arch/arm/boot/compressed/head.S	2006-01-03 03:21:10.000000000 +0000
+++ linux-2.6.15nslu2/arch/arm/boot/compressed/head.S	2006-12-19 00:37:12.000000000 +0000
@@ -118,6 +118,10 @@
 1:		mov	r7, r1			@ save architecture ID
 		mov	r8, #0			@ save r0
 
+		/* Force to be an NSLU2 */
+		mov	r7, #596
+		add	r7, r7, #1
+
 #ifndef __ARM_ARCH_2__
 		/*
 		 * Booting from Angel - need to enter SVC mode and disable
